test/test_mmdecomp.py

```diff
@@ -0,0 +1,222 @@
+# Owner(s): ["module: nn"]
+
+import contextlib
+from functools import partial
+import sys
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.nn.parameter import Parameter
+import unittest
+from unittest.mock import patch, MagicMock, ANY
+import math
+from torch.backends.cuda import sdp_kernel, SDPBackend
+import torch.optim as optim
+from torch.testing._internal.common_dtype import floating_types_and_half
+from torch.testing._internal.common_device_type import instantiate_device_type_tests, onlyCUDA, onlyCPU
+from typing import List, Tuple, Union, Optional
+from torch.testing._internal.common_nn import NNTestCase
+from torch.testing._internal.common_utils import (
+    TEST_FAIRSEQ,
+    run_tests,
+    parametrize,
+    freeze_rng_state,
+    TEST_WITH_CROSSREF,
+    slowTest,
+    set_default_dtype,
+    gradcheck
+)
+
+
+from torch.testing._internal.common_methods_invocations import wrapper_set_seed
+from torch.testing._internal.common_cuda import SM80OrLater, PLATFORM_SUPPORTS_FUSED_SDPA
+
+if TEST_FAIRSEQ:
+    import fairseq.models.transformer as fairseq_transformer
+
+
+@contextlib.contextmanager
+def use_deterministic_algorithims(mode: bool, warn_only: bool):
+    r"""
+    This context manager can be used to temporarily enable or disable deterministic algorithms.
+    Upon exiting the context manager, the previous state of the flag will be restored.
+    """
+    previous_mode: bool = torch.are_deterministic_algorithms_enabled()
+    previous_warn_only: bool = torch.is_deterministic_algorithms_warn_only_enabled()
+    try:
+        torch.use_deterministic_algorithms(mode, warn_only = warn_only)
+        yield {}
+    finally:
+        torch.use_deterministic_algorithms(previous_mode, warn_only = previous_warn_only)
+
+
```

Comment:
Isn't it the same as `torch.testing._internal.DeterministicGuard`? Besides, it's unused, so why not delete it?
```suggestion
```

Reply:
Copy pasta from test_transformer.py ;)